spark/src/main/scala/org/apache/comet/rules/CometExecRule.scala

```diff
@@ -218,6 +226,23 @@ case class CometExecRule(session: SparkSession) extends Rule[SparkPlan] {
         val nativeOp = operator2Proto(cometOp)
         CometScanWrapper(nativeOp.get, cometOp)
 
+      // Handle DataWritingCommandExec specially since it doesn't follow the standard pattern
+      case exec: DataWritingCommandExec =>
+        CometExecRule.writeExecs.get(classOf[DataWritingCommandExec]) match {
```

Comment:
This logic is a bit funky, but I didn't want to spend too much time on this for the initial PR. The logic will change once we add support for fully native inputs (such as `native_datafusion`) anyway.